File: demo/src/main/java/com/bytedance/scenedemo/navigation/performance/PerformanceDemo.java
package com.bytedance.scenedemo.navigation.performance;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.bytedance.scene.Scene;
import com.bytedance.scenedemo.R;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 8/21/18.
 */
public class PerformanceDemo extends Scene {

    public static long startTimestamp = 0;

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return (ViewGroup) inflater.inflate(R.layout.basic_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 0));

        TextView name = getView().findViewById(R.id.name);
        name.setText(getString(R.string.nav_compare_description));

        Button btn = getView().findViewById(R.id.btn);
        btn.setText(getString(R.string.nav_compare_btn_1));
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startTimestamp = System.currentTimeMillis();
                getNavigationScene().startActivity(new Intent(getActivity(), EmptyActivity.class));
            }
        });

        Button btn2 = getView().findViewById(R.id.btn2);
        btn2.setVisibility(View.VISIBLE);
        btn2.setText(getString(R.string.nav_compare_btn_2));
        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startTimestamp = System.currentTimeMillis();
                getNavigationScene().push(EmptyScene.class);
            }
        });
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/benchmark/performance/PerformanceDemo.kt
package com.bytedance.scenedemo.benchmark.performance

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.bytedance.scenedemo.R
import android.widget.TextView
import com.bytedance.scenedemo.benchmark.performance.PerformanceDemo
import android.content.Intent
import android.view.View
import android.widget.Button
import com.bytedance.scene.Scene
import com.bytedance.scene.ktx.navigationScene
import com.bytedance.scenedemo.benchmark.performance.EmptyAppCompatActivity
import com.bytedance.scenedemo.benchmark.performance.EmptyAppCompatScene
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 8/21/18.
 */
class PerformanceDemo : Scene() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.basic_layout, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.setBackgroundColor(ColorUtil.getMaterialColor(resources, 0))
        val name = view.findViewById<TextView>(R.id.name)
        name.text = getString(R.string.nav_compare_description)
        val btn = view.findViewById<Button>(R.id.btn)
        btn.text = getString(R.string.nav_compare_btn_1)
        btn.setOnClickListener {
            startTimestamp = System.currentTimeMillis()
            navigationScene!!.startActivity(Intent(activity, EmptyAppCompatActivity::class.java))
        }
        val btn2 = view.findViewById<Button>(R.id.btn2)
        btn2.visibility = View.VISIBLE
        btn2.text = getString(R.string.nav_compare_btn_2)
        btn2.setOnClickListener {
            startTimestamp = System.currentTimeMillis()
            navigationScene!!.push(EmptyAppCompatScene::class.java)
        }
    }

    companion object {
        var startTimestamp: Long = 0
    }
}